### PR TITLE
Simplify Most Popular/Viewed Footer grid

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Using `grid-auto-flow: column` means we no longer need specific placement of items in the grid, and can remove the logic about when to show top borders.

## Why?

Simplicity is the ultimate sophistication. Also, it aligns with colouring things with the palette!

## Screenshots

| Before | After |
|--------|--------|
| before-dark | after-dark |

### CODE REVIEW

File: dotcom-rendering/src/components/MostPopularFooterGrid.tsx [hunk 1]
```diff
@@ -2,17 +2,17 @@ import { css } from '@emotion/react';
...
 } from '@guardian/source/foundations';
 import { Link } from '@guardian/source/react-components';
+import { palette } from '../palette';
 import type { TrailTabType, TrailType } from '../types/trails';
 import { MostViewedFooterItem } from './MostViewedFooterItem';
 
 const gridContainerStyle = css`
 	display: grid;
-	grid-auto-flow: 1fr;
+	grid-auto-flow: column;
```
Comment (mxdvl) [inline dotcom-rendering/src/components/MostPopularFooterGrid.tsx:15]: This really helps!

File: dotcom-rendering/src/palette.ts [hunk 2]
```diff
@@ -6352,6 +6352,10 @@ const paletteColours = {
 		light: matchActiveTabBorderLight,
 		dark: matchActiveTabBorderDark,
 	},
+	'--most-viewed-description': {
```
Comment (domlander) [inline dotcom-rendering/src/palette.ts:6355]: All other properties in this file look to be extracted out. Could we do the same here for consistency purposes?

File: dotcom-rendering/src/components/MostViewedFooterItem.tsx [hunk 3]
```diff
@@ -2,49 +2,31 @@ import type { SerializedStyles } from '@emotion/react';
...
 import { LinkHeadline } from './LinkHeadline';
 
-const gridItem = (position: number, hasPageSkin: boolean) => {
-	const borderTop = hasPageSkin
-		? css`
-				border-top: 1px solid ${palette('--article-border')};
-		  `
-		: css`
-				/* Below leftCol always set top border */
-				${until.leftCol} {
```
Comment (domlander) [inline dotcom-rendering/src/components/MostViewedFooterItem.tsx:19]: From leftCol, there is a double border below the heading

Screenshot 2024-07-22 at 09 18 57

File: dotcom-rendering/src/palette.ts (hunk 2 above)
Comment (mxdvl) [reply dotcom-rendering/src/palette.ts:6355]: There are [10 other instances of this pattern](https://github.com/guardian/dotcom-rendering/blob/25727ee84d4d1bc377d0800cd6e107aa77d1d540/dotcom-rendering/src/palette.ts#L6723-L6745), but if you’d prefer it extracted I see no objection – you’ll look after this codebase longer than I.

Worth doing the entire extraction in a separate, follow-up PR?

Comment (domlander) [reply dotcom-rendering/src/palette.ts:6355]: It's a very, very small thing so I'd suggest leave it until the next person touches this file

**Pull request merged**